db/migrate/20150702160751_add_user_id_to_articles.rb

```diff
@@ -0,0 +1,40 @@
+class AddUserIdToArticles < ActiveRecord::Migration
+  def up
+    #add user id column
+    add_column :articles, :user_id, :integer
+
+    #converts username to id
+    execute <<-SQL
+      UPDATE articles
+      SET user_id = users.id
+      FROM users
+      WHERE users.username = articles.author
+    SQL
+
+    #drop author column from article
+    remove_column :articles, :author
+
+    #add null constraint to user id
+    change_column_null :articles, :user_id, false
```

Comment:
Should that last argument be `false` or `true` if you want to add a constraint?  Haven't used `change_column_null` before.. but figured it'd be worth mentioning just in case.